lib/cosmos/interfaces/cmd_tlm_server_interface.rb

```diff
@@ -46,37 +55,49 @@ def disconnect
     def read
       while connected?
         begin
-          event = CmdTlmServer.instance.get_limits_event(@limit_id)
-          if event
-            if event[0] == :LIMITS_CHANGE
-              data = event[1]
-              packet ||= System.telemetry.packet("SYSTEM","LIMITS_CHANGE")
-              packet.received_time = Time.now.sys
-              packet.write('PKT_ID',2)
-              packet.write('TARGET', data[0])
-              packet.write('PACKET', data[1])
-              packet.write('ITEM', data[2])
-              # For the first limits change the old_state is nil
-              # so set it to a usable string
-              data[3] = 'UNKNOWN' unless data[3]
-              packet.write('OLD_STATE', data[3])
-              packet.write('NEW_STATE', data[4])
-              @read_count += 1
-              @read_raw_data_time = Time.now
-              @read_raw_data = packet.buffer
-              return packet
+          begin
+            event = CmdTlmServer.instance.get_limits_event(@limit_id, true)
```

Comment:
This is now non-blocking and with the sleeper it means that limits events are processed in batches at 1Hz but I think that's a fair trade off.